### PR TITLE
Unflake TransportWriteThrottle

### PR DESCRIPTION
**Pull request by ali-ince**

### CODE REVIEW

File: community/bolt/src/test/java/org/neo4j/bolt/transport/TransportWriteThrottleTest.java [hunk 1]
```diff
@@ -166,85 +164,122 @@ public void shouldResumeWhenWritableOnceAgain() throws Exception
...
     {
+        TestThrottleLock lockOverride = new TestThrottleLock();
+
         // given
-        TransportThrottle throttle = newThrottleAndInstall( channel );
+        TransportThrottle throttle = newThrottleAndInstall( channel, lockOverride );
         when( channel.isWritable() ).thenReturn( false );
 
-        Future future = Executors.newSingleThreadExecutor().submit( () -> throttle.acquire( channel ) );
+        Thread throttleThread = Executors.defaultThreadFactory().newThread( () -> throttle.acquire( channel ) );
```
Comment (lutovich) [inline community/bolt/src/test/java/org/neo4j/bolt/transport/TransportWriteThrottleTest.java]: Maybe use `OtherThreadRule` class? It allows running things in a different thread and has ability to check thread state.

File: community/bolt/src/test/java/org/neo4j/bolt/transport/TransportWriteThrottleTest.java [hunk 2]
```diff
@@ -166,85 +164,122 @@ public void shouldResumeWhenWritableOnceAgain() throws Exception
...
+    private TransportThrottle newThrottleAndInstall( Channel channel, ThrottleLock lockOverride )
+    {
+        TransportThrottle throttle = newThrottle( lockOverride );
 
         throttle.install( channel );
 
         return throttle;
     }
 
+    public static void awaitWaitingState( Thread thread, long timeoutMillis ) throws TimeoutException
```
Comment (lutovich) [inline community/bolt/src/test/java/org/neo4j/bolt/transport/TransportWriteThrottleTest.java]: This method can probably be replaced with `Predicates#await()`

**Review by lutovich — COMMENTED** [review]
LGTM, couple small comments

File: community/bolt/src/test/java/org/neo4j/bolt/transport/TransportWriteThrottleTest.java (hunk 1 above)
Comment (ali-ince) [reply community/bolt/src/test/java/org/neo4j/bolt/transport/TransportWriteThrottleTest.java]: Thanks for pointing out, didn't know that.

**Pull request merged**